crypto/x509/x509_lu.c

```diff
@@ -93,22 +93,18 @@ int X509_LOOKUP_by_subject_ex(X509_LOOKUP *ctx, X509_LOOKUP_TYPE type,
                               const X509_NAME *name, X509_OBJECT *ret,
                               OSSL_LIB_CTX *libctx, const char *propq)
 {
-    if (ctx->skip
-        || ctx->method == NULL
-        || (ctx->method->get_by_subject == NULL
-            && ctx->method->get_by_subject_ex == NULL))
+    if (ctx->skip || ctx->method == NULL
+           || ctx->method->get_by_subject_ex == NULL)
         return 0;
-    if (ctx->method->get_by_subject_ex != NULL)
-        return ctx->method->get_by_subject_ex(ctx, type, name, ret, libctx,
-                                              propq);
-    else
-        return ctx->method->get_by_subject(ctx, type, name, ret);
+    return ctx->method->get_by_subject_ex(ctx, type, name, ret, libctx, propq);
 }
 
 int X509_LOOKUP_by_subject(X509_LOOKUP *ctx, X509_LOOKUP_TYPE type,
                            const X509_NAME *name, X509_OBJECT *ret)
 {
-    return X509_LOOKUP_by_subject_ex(ctx, type, name, ret, NULL, NULL);
+    if (ctx->skip || ctx->method == NULL || ctx->method->get_by_subject == NULL)
```

Comment:
Please revert this change also.. The intention was to call the new function with NULL, NULL..

Reply:
Done.